uaclient/data_types.py

```diff
@@ -114,12 +114,13 @@ def from_value(val: Any) -> List:
 
 
 def data_list_to_list(
-    val: List[Union["DataObject", list, str, int, bool]]
+    val: List[Union["DataObject", list, str, int, bool]],
+    keep_none: bool = True,
 ) -> list:
     new_val = []  # type: list
     for item in val:
         if isinstance(item, DataObject):
-            new_val.append(item.to_dict())
+            new_val.append(item.to_dict(keep_none))
         elif isinstance(item, list):
             new_val.append(data_list_to_list(item))
```

Comment:
keep_none propagation is needed here too
```suggestion
            new_val.append(data_list_to_list(item, keep_none))
```

Reply:
Thanks for that catch.